src/libraries/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -2167,6 +2167,115 @@ public async Task SendHttpPostRequest_When100ContinueSet_ReceivedByServer(bool e
+                    tcs.SetResult();
+                    Assert.NotNull(exception.Response);
+                    using (var responseStream = exception.Response.GetResponseStream())
+                    {
+                        var buffer = new byte[10];
+                        int readLen = responseStream.Read(buffer, 0, buffer.Length);
+                        Assert.Equal(5, readLen);
+                        Assert.Equal(new string('a', 5), Encoding.UTF8.GetString(buffer[0..readLen]));
+                    }
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(
+                        async (client) =>
+                        {
+                            await client.SendResponseAsync(statusCode: HttpStatusCode.BadRequest, content: new string('a', 10));
+                            await tcs.Task;
+                        });
+                });
+            }, (this is HttpWebRequestTest_Async).ToString()).Dispose();
+        }
+
+        [Fact]
+        public void HttpWebRequest_SetProtocolVersion_Success()
+        {
+            HttpWebRequest request = WebRequest.CreateHttp(Configuration.Http.RemoteEchoServer);
+
+            request.ProtocolVersion = HttpVersion.Version10;
+            Assert.Equal(HttpVersion.Version10, request.ServicePoint.ProtocolVersion);
+
+            request.ProtocolVersion = HttpVersion.Version11;
+            Assert.Equal(HttpVersion.Version11, request.ServicePoint.ProtocolVersion);
+        }
+
+        [Fact]
+        public async Task SendHttpRequest_BindIPEndPoint_Success()
+        {
+            TaskCompletionSource tcs = new TaskCompletionSource();
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.ServicePoint.BindIPEndPointDelegate =
+                        (sp, ep, retryCount) => { return new IPEndPoint(IPAddress.Loopback, 27277); };
+                    using (var response = (HttpWebResponse)await GetResponseAsync(request))
+                    {
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                    }
+                    tcs.SetResult();
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(
+                        async (client) =>
+                        {
+                            var ipEp = (IPEndPoint)client.Socket.RemoteEndPoint;
+                            Assert.Equal(27277, ipEp.Port);
+                            await client.SendResponseAsync();
+                            await tcs.Task;
+                        });
+                });
+        }
+
+        [Fact]
+        public async Task SendHttpRequest_BindIPEndPoint_Throws()
```

Comment:
This should use `RemoteExecutor` as well since it's modifying the shared ServicePoint